components/formats-gpl/src/loci/formats/in/UnisokuReader.java

```diff
@@ -206,7 +211,7 @@ else if (key.startsWith(":y_data ->")) {
             pixelSizeY = Double.parseDouble(v[2]) - Double.parseDouble(v[1]);
             pixelSizeY /= getSizeY();
             if (unit.equals("nm")) {
-              pixelSizeY /= 1000;
+              pixelSizeYUnit = UNITS.NM;
```

Comment:
Well you can pass the String unit to FormatTools also (FormatTools.getPhysicalSizeX(value, unit)) and it will do UnitsLength.fromString(unit), which is essentially the same thing. The UNITS class isn't an enum so there isn't an easy way of converting the String without a large if else.